libgalois/CMakeLists.txt

```diff
@@ -43,55 +43,54 @@ if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
   list(APPEND sources src/HWTopoDarwin.cpp)
 else()
   include(CheckSchedSetAffinity)
-  if (SCHED_SETAFFINITY_FOUND)
-    add_definitions(-DGALOIS_USE_SCHED_SETAFFINITY)
-  elseif (USE_STRICT_CONFIG)
-    message(FATAL_ERROR "Need sched_setaffinity")
+  if (NOT SCHED_SETAFFINITY_FOUND)
+    if (USE_STRICT_CONFIG)
+      message(FATAL_ERROR "Need sched_setaffinity")
+    endif()
   endif()
-
   list(APPEND sources src/HWTopoLinux.cpp)
 endif()
 
-add_library(galois_shmem ${sources})
+add_library(galois_shmem_obj OBJECT ${sources})
```

Comment:
Why the distinction between galois_shmem and galois_shmem_obj?

Reply:
Not my addition, but libcusp uses this to, I presume, build a fully resolved static library (and not depend on galois_shmem). Object libraries are the only cmake-portable way to do this. My change was to optimize this by actually using the compiled object files for both galois_shmem and galois_shmem_obj, rather than redundantly compiling them separately.

I think going forward I'd love to remove these weird edge cases (e.g., avoid object libraries altogether or use object libraries everywhere and have a final "library packaging" phase).

Reply:
Got it. Thanks for the clarification. My preference is generally for shared objects everywhere, but I'm sure we can figure something sane out here.